mdn/kuma
# Update mobile search interaction

Changes
- Add toggle button that handles showing and hiding the search form
- Update styles so it looks like the new mobile design, but keeping the tablet and desktop the same 

Closes #7011 

Note: This PR does not address item 5 in the ticket, which is adding the back button. Mustafa said he was fine with leaving it out, and our current router does not appear to give us the ability to go back in history. I'm going to take a look at the router code tomorrow, as there seem to be functions that indicate this was possible in the past, to see how difficult this would be. But any how, it will be a separate PR if it does end up happening. 

To test: 
1. Using a small screen width, go to any of the React pages. Click on magnifying glass in top right corner:
<img width="377" alt="Screen Shot 2020-05-11 at 10 40 55 PM" src="https://user-images.githubusercontent.com/650/81632856-43a08100-93d9-11ea-9227-cf32daf29b46.png">

2. Ensure that scrolling down covers content on the page: 
<img width="381" alt="Screen Shot 2020-05-11 at 10 41 18 PM" src="https://user-images.githubusercontent.com/650/81632884-53b86080-93d9-11ea-8cea-c65203da35c1.png">

3. Type a search term and press "Enter", should be taken to search page with search input pre-filled:
<img width="376" alt="Screen Shot 2020-05-11 at 10 50 54 PM" src="https://user-images.githubusercontent.com/650/81633091-e527d280-93d9-11ea-88d2-a02e9cc5d02a.png">

4. Make window size larger, should see the rounded search input that is on the production site: 
<img width="1174" alt="Screen Shot 2020-05-11 at 10 52 07 PM" src="https://user-images.githubusercontent.com/650/81633158-11dbea00-93da-11ea-9fd2-e599ef497cf5.png">
